packages/ember-data/lib/serializers/rest-serializer.js

```diff
@@ -40,8 +40,12 @@ function coerceId(id) {
 
   ```js
   App.ApplicationSerializer = DS.RESTSerializer.extend({
-    keyForAttribute: function(attr) {
-      return Ember.String.underscore(attr).toUpperCase();
+    keyForAttribute: function(attr, method) {
+      if (method === 'serialize') {
+        return Ember.String.underscore(attr).toUpperCase();
+      } else {
+        return Ember.String.camelize(attr.toLowerCase());
+      }
```

Comment:
The original example would work fine if the data is never saved back to the server. If it is, though, we do need to perform the reverse conversion. Perhaps I should instead expand the text above and and a bit clarifying this reverse procedure?